src/validations/mta/mtaValidations.ts

```diff
@@ -116,35 +117,47 @@ async function getMtaDiagnostics(
   const mtaPath = resolve(modulePath, MTA_YAML);
   const devMtaExtPath = resolve(modulePath, DEV_MTA_EXT);
 
-  // no mta.yaml
-  if ((await pathExists(mtaPath)) === false) {
+  // no mta.yaml. clear the previous diagnostics and return no diagnostics
+  if (!(await pathExists(mtaPath))) {
     return diagnosticsByFile;
   }
 
   // if dev.mtaext exist => validate with dev.mtaext
-  const devMtaExts: string[] | undefined =
-    (await pathExists(devMtaExtPath)) === true ? [devMtaExtPath] : undefined;
+  const devMtaExts: string[] | undefined = (await pathExists(devMtaExtPath))
+    ? [devMtaExtPath]
+    : undefined;
   const mta = new Mta(modulePath, false, devMtaExts); // temp file is not relevant in our scenario
 
   const validationRes: Record<string, mta.Issue[]> = await mta.validate();
 
   for (const filePath of keys(validationRes)) {
-    diagnosticsByFile[filePath] = diagnosticsByFile[filePath] ?? [];
-
     // iterate validation issues a single file
-    for (const issue of validationRes[filePath]) {
-      const position = new Position(issue.line, issue.column);
-      diagnosticsByFile[filePath].push({
-        source: "MTA", // Should be synchronized with package.json
-        message: issue.message,
-        range: new Range(position, position),
-        severity:
-          issue.severity === "error"
-            ? DiagnosticSeverity.Error
-            : DiagnosticSeverity.Warning,
-      });
-    }
+    diagnosticsByFile[filePath] = map<mta.Issue, Diagnostic>(
+      validationRes[filePath],
+      (issue) => {
+        const position = new Position(
+          convertMtaIssueCoordinateToEditorCoordinate(issue.line),
+          convertMtaIssueCoordinateToEditorCoordinate(issue.column)
+        );
+        return {
+          source: "MTA", // Should be synchronized with package.json
+          message: issue.message,
+          range: new Range(position, position),
+          severity: DiagnosticSeverity.Warning,
+        };
+      }
+    );
   }
 
   return diagnosticsByFile;
 }
+
+function convertMtaIssueCoordinateToEditorCoordinate(
+  coordinate: number
+): number {
+  const number = Number(coordinate) - 1;
```

Comment:
Please add a comment that this conversion is necessary because mta issues have 1-based lines and columns (unless no line/column number is available and then it's 0), and vscode has 0-based lines and columns